jetty-core/jetty-client/src/main/java/org/eclipse/jetty/client/Connection.java

```diff
@@ -63,4 +64,13 @@ default SocketAddress getRemoteSocketAddress()
     {
         return null;
     }
+
+    /**
+     * @return the {@link SSLSession} associated with the connection, or {@code null}
+     * if the connection is not secure or the {@link SSLSession} is not available.
+     */
+    default SSLSession getSSLSession()
```

Comment:
Is there any benefit of returning an EndPoint.SslSessionData instead?  It can provide more information or just the ID if the actual session is abstracted away.

Reply:
I don't see much benefit? `SSLSession` is from the JDK, so it is well known.
Also, this is the client, and it does not need the number of tricks the server should support when it's behind a reverse proxy (e.g. support non-standard HTTP headers that a proxy may add that specify the cipher suite, etc.)

Reply:
@gregw actually, I changed my mind.

The reason being HTTP/3, which won't have an `SSLSession`, but there could be enough information to return the cipher suite and the peer certificates.